provider/vsphere/internal/vsphereclient/client.go

```diff
@@ -456,23 +458,65 @@ func (c *Client) destroyVM(
 
 func (c *Client) cloneVM(
 	ctx context.Context,
+	args CreateVirtualMachineParams,
 	srcVM *object.VirtualMachine,
 	dstName string,
 	vmFolder *object.Folder,
+	datacenter *object.Datacenter,
 	taskWaiter *taskWaiter,
 ) (*object.VirtualMachine, error) {
+	configSpec := &types.VirtualMachineConfigSpec{}
+
+	if args.Constraints.HasCpuCores() {
```

Comment:
So we were ignoring constraints?

Reply:
No. They didn't appear here as they were being dealt with in a different area of the code.